app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadAdapter.kt

```diff
@@ -476,10 +481,112 @@ class ThreadAdapter(
             reloadVisibleWebView()
 
             distantImagesAlert.isGone = true
-            hideAlertGroupIfNoneDisplayed()
         }
     }
 
+    //region Spam
+    private enum class SpamAction {
+        MoveToSpam, EnableFilter, Unblock, None
+    }
+
+    private data class SpamData(val spamAction: SpamAction, val description: String = "", val action: String = "")
+
+    private fun MessageViewHolder.bindSpam(message: Message) = with(binding) {
+        val firstExpeditor = message.from.first()
+        val spamData = context.getSpamBannerData(spamAction = getSpamBannerAction(message), emailToUnblock = firstExpeditor.email)
+
+        if (spamData.spamAction == SpamAction.None) {
+            spamAlert.isVisible = false
+        } else {
+            spamAlert.isVisible = true
+            spamAlert.setDescription(spamData.description)
+            spamAlert.setAction1Text(spamData.action)
+            spamAlert.onAction1 { spamActionButton(spamData, message, firstExpeditor) }
+        }
+
+        hideAlertGroupIfNoneDisplayed()
```

Comment:
Why did you remove this line from the `distantImagesAlert.onAction1 { … }` click listener in `bindAlerts()` ?
I think we still need it there.

Reply:
Done.

Reply:
Done.